.github/workflows/node.js.yml

```diff
@@ -15,21 +15,19 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [12.x, 14.x, 16.x]
+        node-version: [14.x, 16.x]
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
 
     steps:
       - name: git checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
           cache: 'npm'
       - name: Start MongoDB
         uses: supercharge/mongodb-github-action@1.6.0
-      - run: npm i mongoose@^5.8.10
+      - run: npm i mongoose@6.x
       - run: npm run lint
-      - run: npm run test ci
-      - run: npm install coveralls@~2.11.0
-      - run: npm run report-coverage
```

Comment:
Reporting to coveralls also caused issues and I'm not sure it's worth maintaining that at this point. Anecdotally, I feel like it's very uncommon for open-source projects to use that service.